server/jms/src/main/java/org/hawkular/apm/server/jms/RetryCapableMDB.java

```diff
@@ -95,8 +98,7 @@ public void onMessage(Message message) {
             process(tenantId, items, retryCount);
 
         } catch (Exception e) {
-            // TODO: Handle nak of JMS message?
```

Comment:
The problem is how to classify what failures should result in retry - that should/could be determined by the specific processor that knows the technology and determines whether an exception should be thrown into the RetryCapableMDB - as has been done with the Elasticsearch case. Initially the response failures were not triggering the retry - but now, because I added the throw, it will result in a retry.

So the decision whether to expose the exception is in the specific processor implementation.